### DISCUSSION TITLE
Is opentelemetry going to be a replacement of APM tool?

### DISCUSSION THREAD

**Discussion by goldyliang** [open-telemetry/opentelemetry-java-instrumentation#3857] [Q&A]
The Java APM (application performance monitoring) tools have some overlap vs the auto Java agent here especially tracing. A few APM tools support decent tracing (not necessary distributed though) which opentelmetry Java agent is catching up. On the other hand APM mostly supports global profiling and profiling within an entry of a trace (called event here?) 

Is opentemetey Java agent supposed to have the capability of APM as well (one day) or will it just focuse on metrics and tracing?

We are using glowroot (one AMAZING Java APM) now and I am very curious about the fate of glowroot as I see @trask is both the author of glowroot and active contributor of this Java agent, and Trask seems to be working on this agent other than maintaining glowroot most recently.

**Comment by trask** [comment]
hey @goldyliang! There's definitely [interest in profiling](https://github.com/open-telemetry/oteps/issues/139), and I would expect to see the OpenTelemetry Java agent support profiling in the future.
